backend/tests/test_middleware.py

```
@@ -0,0 +1,612 @@
+
+            assert exc_info.value.status_code == status.HTTP_401_UNAUTHORIZED
+
+    @pytest.mark.asyncio
+    async def test_valid_tenant_extraction(self, middleware, mock_request):
+        """Test successful tenant extraction and context setting."""
+        tenant_id = uuid.uuid4()
+        mock_request.url.path = "/api/v1/projects"
+
+        with patch.object(middleware, '_extract_tenant_from_request', return_value=tenant_id):
+            call_next = AsyncMock(return_value=Response())
+            response = await middleware.dispatch(mock_request, call_next)
```

Ruff (0.13.1)

Local variable `response` is assigned to but never used

Remove assignment to unused variable `response`

(F841)